app/src/display/widgets/wpm_status.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/display/widgets/wpm_status.h>
+#include <zmk/events/wpm_state_changed.h>
+#include <zmk/event_manager.h>
+#include <zmk/endpoints.h>
+#include <zmk/wpm.h>
+
+static sys_slist_t widgets = SYS_SLIST_STATIC_INIT(&widgets);
+static lv_style_t label_style;
+
+static bool style_initialized = false;
+
+void wpm_status_init() {
+    if (style_initialized) {
+        return;
+    }
+
+    style_initialized = true;
+    lv_style_init(&label_style);
+    lv_style_set_text_color(&label_style, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+    lv_style_set_text_font(&label_style, LV_STATE_DEFAULT, &lv_font_montserrat_12);
+    lv_style_set_text_letter_space(&label_style, LV_STATE_DEFAULT, 1);
+    lv_style_set_text_line_space(&label_style, LV_STATE_DEFAULT, 1);
+}
+
+void set_wpm_symbol(lv_obj_t *label, int wpm) {
+    char text[4] = {};
+
+    LOG_DBG("WPM changed to %i", wpm);
+    sprintf(text, "%i ", wpm);
+
+    lv_label_set_text(label, text);
+}
+
+int zmk_widget_wpm_status_init(struct zmk_widget_wpm_status *widget, lv_obj_t *parent) {
+    wpm_status_init();
+    widget->obj = lv_label_create(parent, NULL);
+    lv_obj_add_style(widget->obj, LV_LABEL_PART_MAIN, &label_style);
+    lv_label_set_align(widget->obj, LV_LABEL_ALIGN_RIGHT);
+
+    lv_obj_set_size(widget->obj, 40, 15);
+    set_wpm_symbol(widget->obj, 0);
+
+    sys_slist_append(&widgets, &widget->node);
+
+    return 0;
+}
+
+lv_obj_t *zmk_widget_wpm_status_obj(struct zmk_widget_wpm_status *widget) { return widget->obj; }
+
+int wpm_status_listener(const struct zmk_event_header *eh) {
+    struct wpm_state_changed *ev = cast_wpm_state_changed(eh);
+    struct zmk_widget_wpm_status *widget;
+    SYS_SLIST_FOR_EACH_CONTAINER(&widgets, widget, node) { set_wpm_symbol(widget->obj, ev->state); }
+    return 0;
+}
+
+ZMK_LISTENER(widget_wpm_status, wpm_status_listener)
+ZMK_SUBSCRIPTION(widget_wpm_status, wpm_state_changed);
```

Comment:
To fixup / reconcile this block with the current main sources you'll need to change it to the following:

```
int wpm_status_listener(const zmk_event_t *eh) {
    struct zmk_wpm_state_changed *ev = as_zmk_wpm_state_changed(eh);
    struct zmk_widget_wpm_status *widget;
    SYS_SLIST_FOR_EACH_CONTAINER(&widgets, widget, node) { set_wpm_symbol(widget->obj, ev->state); }
    return 0;
}

ZMK_LISTENER(widget_wpm_status, wpm_status_listener)
ZMK_SUBSCRIPTION(widget_wpm_status, zmk_wpm_state_changed);
```

Reply:
Thanks.  Didn't see that when I was building locally but did when I built my config.  I'll look into that tomoz.  I think that's it :)